### PR TITLE
fix: Updating azure_logcollector.go to reflect containerd runtime on Windows

### PR DESCRIPTION
**Pull request by marosset**
Signed-off-by: Mark Rossetti <marosset@microsoft.com>

 

 
**What type of PR is this?**
/kind cleanup
/kind bug

**What this PR does / why we need it**:
This PR updates log collection for Windows nodes to gather logs assuming containerd is the container runtime, not docke.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note

```

### CODE REVIEW

File: test/e2e/azure_logcollector.go [hunk 1]
```diff
@@ -237,13 +237,9 @@ func windowsK8sLogs(execToPathFn func(outputFileName string, command string, arg
...
 			"Get-WinEvent", "-LogName Microsoft-Windows-Hyper-V-Compute-Operational | Select-Object -Property TimeCreated, Id, LevelDisplayName, Message | Sort-Object TimeCreated | Format-Table -Wrap -Autosize",
 		),
-		execToPathFn(
-			"docker.log",
-			"get-eventlog", "-LogName Application -Source Docker | Select-Object Index, TimeGenerated, EntryType, Message | Sort-Object Index | Format-Table -Wrap -Autosize",
-		),
 		execToPathFn(
 			"containers.log",
-			"docker", "ps -a",
+			"ctr.exe", "-n k8s.io containers list",
```
Comment (jsturtevant) [inline test/e2e/azure_logcollector.go:242]: We should include `ctr task list` as well.  There have been times when there were containers left over but tasks where removed.

Comment (marosset) [reply test/e2e/azure_logcollector.go:242]: updated

File: test/e2e/azure_logcollector.go [hunk 2]
```diff
@@ -238,12 +238,12 @@ func windowsK8sLogs(execToPathFn func(outputFileName string, command string, arg
...
 		execToPathFn(
-			"docker.log",
-			"get-eventlog", "-LogName Application -Source Docker | Select-Object Index, TimeGenerated, EntryType, Message | Sort-Object Index | Format-Table -Wrap -Autosize",
+			"containers.log",
+			"ctr.exe", "-n k8s.io containers list",
 		),
 		execToPathFn(
-			"containers.log",
-			"docker", "ps -a",
+			"hcsshim-taskss.log",
```
Comment (jsturtevant) [inline test/e2e/azure_logcollector.go:245]:
```suggestion
			"containerd-tasks.log",
```

File: test/e2e/azure_logcollector.go [hunk 3]
```diff
@@ -238,12 +238,12 @@ func windowsK8sLogs(execToPathFn func(outputFileName string, command string, arg
 			"Get-WinEvent", "-LogName Microsoft-Windows-Hyper-V-Compute-Operational | Select-Object -Property TimeCreated, Id, LevelDisplayName, Message | Sort-Object TimeCreated | Format-Table -Wrap -Autosize",
 		),
 		execToPathFn(
-			"docker.log",
-			"get-eventlog", "-LogName Application -Source Docker | Select-Object Index, TimeGenerated, EntryType, Message | Sort-Object Index | Format-Table -Wrap -Autosize",
+			"containers.log",
```
Comment (jsturtevant) [inline test/e2e/azure_logcollector.go:241]:
```suggestion
			"containerd-containers.log",
```

**Comment by jsturtevant** [comment]
Files are being read as expected: https://support.microsoft.com/en-us/topic/windows-10-and-windows-server-2019-update-history-725fc2e1-4443-6831-a5ca-51ff5cbcb059

/lgtm

**Pull request merged**